server/routes/bookingsRoute.js

```diff
@@ -47,4 +49,52 @@ router.post('/students/:student_id/bookings', lectureValidation.checkLecture(),v
         res.status(400).json(error);
     }
 })
+
+
+/**
+ * @swagger
+ * //students/{student_id}/delete_booking:
+ *  post:
+ *    tags:
+ *      - students
+ *    summary: "Delete a new booking"
+ *    description: "A student can delete a lecture "
+ *    parameters:
+ *      - in: path
+ *        name: student_id
+ *        required: true
+ *        schema:
+ *          type: integer
+ *          format: int64
+ *    responses:
+ *       "201 and num=1":
+ *         description: "Successful deletion"
+ *         schema:
+ *           type: "object"
+ *       "201 and num=0":
+ *         description: "Nothing to delete"
+ *         schema:
+ *           type: "object"
+ *       "400":
+ *         description: "Invalid status value"
+ *    security:
+ *     - petstore_auth:
+ *       - "write:pets"
+ *       - "read:pets"
+ */
+router.post('/students/:student_id/delete_booking', lectureValidation.checkLecture(),validator, async(req,res) =>{
```

Comment:
The route reported in the documentation for this functionality is `DELETE /students/:student_id/lectures/:lecture_id`, which is a little more restful in my opinion.
 
There is no definitively right choice, we just need to keep it consistent with the docs.

Reply:
Yes, you are right. I've changed the params but before committing I have a question: since it is not a real delete, but only an update of the "deleted_at" field, should we change the documentation in "PUT", or "POST"?

Reply:
I think we should keep `DELETE` as from the client point of view that is a real cancelation of the booking. It is not aware of how the server performs it.